### ISSUE TITLE
[Delta][ERROR]: Deltalog for customer_entity is not installed

### ISSUE THREAD

**Issue by milanstojanov** [magento/data-migration-tool#181]
Can someone please explain why in delta step, the tool is looking for tables with **m2_cl_** prefix:

https://github.com/magento/data-migration-tool/blob/master/src/Migration/ResourceModel/Source.php#L210

I don't have any tables starting with **m2_cl_**, neither in my origianal M1 db, nor in migrated one.

One line that may also be interesting is this:
https://github.com/magento/data-migration-tool/blob/master/etc/ce-to-ce/1.7.0.2/map.xml.dist#L69

Anyhow, I successfully managed to run first two steps, but I get the following error instantaneously after I run the third step:

**Deltalog for customer_entity is not installed**

**Comment by victor-v-rad** [comment]
m2_cl_* tables are for Delta mode. Migration tool should installed it in Data mode. Make sure the tool have rights to create thease tables in Magento 1 DB also look for Customer Step in your config.xml it should be present there.

**Comment by milanstojanov** [comment]
Thanks for your comment, it makes sense now.

BTW - Would it be possible to configure DB migration tool to migrate only particular website (and data related to it)? For example, I have a case when original M1 has 4 websites, but only one is to be migrated to M2. I am wondering if I can define somewhere that I need only website ID=x migrated?

My current process (which I'm still testing) is that I migrate whole M1 database, and then delete unnecessary data from migrated M2 DB. I will test today how DELTA works in that case, but I am looking for more sophisticated process. 

And lastly - since I am in testing process, I always dump original M1 DB and then start migration from the dump (not from the live database). From your comment, I see it is required to run the migration against LIVE db. But is it 100% safe? In other words - it can't break anything?

**Comment by victor-v-rad** [comment]
>  to migrate only particular website

It was designed to migrate all at once.

> it is required to run the migration against LIVE 

If there is no necessity in Delta mode you can run on dump of a store for safety reasons (though I have not encountered complaints in this regard so far). Otherwise you can use your DB slave replica of Magento 1

**Comment by milanstojanov** [comment]
Hm, slave replica usually doesn't have write permissions, which may be a problem since the tool will try to create new tables...

Thanks very much for your responses, I will see what my next steps are.

**Comment by MohanaKrishnan7** [comment]
@victor-v-rad Please explain how to make sure the tool has rights to create these tables in Magento 1 DB ?. 

And what do you mean by "look for Customer Step in your config.xml it should be present there." which config.xml file ?. which line ?. 

PLEASE BE MORE CLEAR IN YOUR ANSWERS.

**Comment by sohrobdevise** [comment]
@MohanaKrishnan7 
I am experiencing the same issue. Have you been able to resolve this?

**Comment by MohanaKrishnan7** [comment]
> @MohanaKrishnan7
> I am experiencing the same issue. Have you been able to resolve this?

Hi Sohrobdevise,

When you RUN the DATA migration log tables will be created in your Magento 1 database. So when you RUN the DELTA migration (which migrates the changes) will check for these tables to update the changes happened in Magento 1 database. So if have RUN the DATA migration in a Magento 1 database and RUN the DELTA migration in another database Magento 2 (which does not have these log tables m2_cl_*) the above-mentioned error will occur.

Once I got this error I have done all the migration steps again in the production (live) Magento 1 database itself. So I was able to complete the migration.

Thanks and Regards,
MohanaKrishnan
